charts/vault-operator/example-terraform.tf

```diff
@@ -0,0 +1,128 @@
+provider "helm" {
+  kubernetes {
+    load_config_file = true
+
+    cluster_ca_certificate = file("~/.kube/cluster-ca-cert.pem")
+  }
+}
+
+locals {
+  kubevault_release_name = "vault-operator"
+  kubevault_namespace    = "kube-system"
+}
+
+locals {
+  kubevault_chart_name = "vault-operator"
+
+  # re-implement: https://github.com/kubevault/operator/blob/0.3.0/charts/vault-operator/templates/_helpers.tpl#L9-L20
+  # in hcl
+  kubevault_release_fullname = length(regexall("\\Q${local.kubevault_chart_name}\\E", local.kubevault_release_name)) != 0 ? trimsuffix(substr(local.kubevault_release_name, 0, 63), "-") : trimsuffix(substr("${local.kubevault_release_name}-${local.kubevault_chart_name}", 0, 63), "-")
+}
+
+data "helm_repository" "appscode" {
+  name = "appscode"
+  url  = "https://charts.appscode.com/stable"
+}
+
+resource "tls_private_key" "kubevault_ca" {
+  algorithm = "RSA"
+  rsa_bits  = 2048
+}
+
+resource "tls_self_signed_cert" "kubevault_ca" {
+  key_algorithm   = tls_private_key.kubevault_ca.algorithm
+  private_key_pem = tls_private_key.kubevault_ca.private_key_pem
+
+  subject {
+    common_name = "ca"
+  }
+
+  validity_period_hours = 87600
+  set_subject_key_id    = false
+
+  is_ca_certificate = true
+  allowed_uses = [
+    "digital_signature",
+    "key_encipherment",
+    "cert_signing",
+    "server_auth",
+    "client_auth"
+  ]
+}
+
+resource "tls_private_key" "kubevault_server" {
+  algorithm = "RSA"
+  rsa_bits  = 2048
+}
+
+resource "tls_cert_request" "kubevault_server" {
+  key_algorithm   = tls_private_key.kubevault_server.algorithm
+  private_key_pem = tls_private_key.kubevault_server.private_key_pem
+
+  subject {
+    common_name = local.kubevault_release_fullname
+  }
+
+  dns_names = [
+    "${local.kubevault_release_fullname}.${local.kubevault_namespace}",
+    "${local.kubevault_release_fullname}.${local.kubevault_namespace}.svc",
+    "${local.kubevault_release_fullname}.${local.kubevault_namespace}.svc.cluster.local"
```

Comment:
Also here?

Reply:
If you insist - I didn't forget  – I was thinking like "well if you are creating the certs for the chart, you are taking on the responsibility of making sure its right";

but sure. one sec.